internal/services/storage/storage_account_customer_managed_key_resource.go

```diff
@@ -49,13 +49,21 @@ func resourceStorageAccountCustomerManagedKey() *pluginsdk.Resource {
 
 			"key_vault_id": {
 				Type:     pluginsdk.TypeString,
-				Required: true,
+				Optional: true,
 				ValidateFunc: validation.Any(
 					// Storage Account Customer Managed Keys support both Key Vault and Key Vault Managed HSM keys:
 					// https://learn.microsoft.com/en-us/azure/storage/common/customer-managed-keys-overview
 					keyVaultValidate.VaultID,
 					keyVaultValidate.ManagedHSMID,
 				),
+				ExactlyOneOf: []string{"key_vault_id", "key_vault_uri"},
+			},
+
+			"key_vault_uri": {
```

Comment:
Assuming with the right role being assigned to the cross tenant app (from the provisioner), the provisioner should has the permission to read the keyvaults and get its URI by its ID, so we can eliminate this new property?

Reply:
I don't believe caller has any access to that Key Vault. The ad app is using a separate federated identity to access it. Using `id` would required sharing additional informations between tenants i.e. `subscription ID` and `resource group name` which isn't necessary for the setup.

This somehow mirrors Azure's UI where you can either select the key vault by id or using the URI.